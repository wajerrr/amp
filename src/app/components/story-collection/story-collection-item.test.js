import React from 'react';
import renderer from 'react-test-renderer';
import StoryCollectionItem from './story-collection-item';

const dummyData = {
  id: '/content/ob99k13ket8ndbhstm7boctldn8joq1o',
  channel: {
    headline: 'About us',
    url: {
      canonical: 'https://www.economist.com/help/about-us',
      __typename: 'URL',
    },
    __typename: 'Content',
  },
  headline: 'Who has the right to judge Americans?',
  subheadline: 'The Economist explains',
  description:
    'A minor fraud case provokes a Supreme Court challenge to the expanding power of administrative agencies in government',
  byline: 'T.E. | NEW YORK',
  url: {
    canonical:
      'https://www.economist.com/the-economist-explains/2018/05/15/who-has-the-right-to-judge-americans',
    __typename: 'URL',
  },
  image: {
    main: {
      width: 1280,
      height: 720,
      url: {
        canonical:
          'https://www.economist.com/sites/default/files/20180512_BLP509.jpg',
        __typename: 'URL',
      },
      __typename: 'Content',
    },
    __typename: 'Media',
  },
  __typename: 'Content',
};

describe('StoryCollectionItem', () => {
  it('should match snapshot without className', () => {
    const tree = renderer
      .create(<StoryCollectionItem data={dummyData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with className', () => {
    const tree = renderer
      .create(<StoryCollectionItem className="class" data={dummyData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
