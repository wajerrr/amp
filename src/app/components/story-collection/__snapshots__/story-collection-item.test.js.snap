// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryCollectionItem should match snapshot with className 1`] = `
<li
  className="class"
>
  <a
    className="css-rfvsuz-StyledLinkWrapper ercjopt1"
    href="https://www.economist.com/the-economist-explains/2018/05/15/who-has-the-right-to-judge-americans"
  >
    <div
      className="css-xpxzo4-StyledImgContainer ercjopt2"
    >
      <amp-img
        alt="About us"
        height={720}
        layout="responsive"
        src="https://www.economist.com/sites/default/files/20180512_BLP509.jpg"
        width={1280}
      />
    </div>
    <div
      className="css-1bn1ahj-StyledTextContainer ercjopt0"
    >
      <h3
        className="css-m4ovun-StyledHeader e10thz6k2"
      >
        <span
          className="css-zdexlw-StyledSubHeadline-subHeadlineStyles e10thz6k1"
        >
          The Economist explains
        </span>
        <span
          className="css-1wbs5nn-StyledHeadline-headlineStyles e10thz6k0"
        >
          Who has the right to judge Americans?
        </span>
      </h3>
    </div>
  </a>
  <a
    className="ercjopt3 css-1cesb30-StyledLink-StyledSectionLink-StyledSectionLink e1pwg3wv0"
    href="https://www.economist.com/help/about-us"
  >
    About us
    <span
      className={undefined}
    >
      <svg
        version="1.1"
        viewBox="0 0 48 48"
        x="0px"
        xmlSpace="preserve"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <polygon
          points="17.6,14.8 20.4,12 32.4,24 20.4,36 17.6,33.2 26.7,24 	"
        />
      </svg>
    </span>
  </a>
</li>
`;

exports[`StoryCollectionItem should match snapshot without className 1`] = `
<li
  className=""
>
  <a
    className="css-rfvsuz-StyledLinkWrapper ercjopt1"
    href="https://www.economist.com/the-economist-explains/2018/05/15/who-has-the-right-to-judge-americans"
  >
    <div
      className="css-xpxzo4-StyledImgContainer ercjopt2"
    >
      <amp-img
        alt="About us"
        height={720}
        layout="responsive"
        src="https://www.economist.com/sites/default/files/20180512_BLP509.jpg"
        width={1280}
      />
    </div>
    <div
      className="css-1bn1ahj-StyledTextContainer ercjopt0"
    >
      <h3
        className="css-m4ovun-StyledHeader e10thz6k2"
      >
        <span
          className="css-zdexlw-StyledSubHeadline-subHeadlineStyles e10thz6k1"
        >
          The Economist explains
        </span>
        <span
          className="css-1wbs5nn-StyledHeadline-headlineStyles e10thz6k0"
        >
          Who has the right to judge Americans?
        </span>
      </h3>
    </div>
  </a>
  <a
    className="ercjopt3 css-1cesb30-StyledLink-StyledSectionLink-StyledSectionLink e1pwg3wv0"
    href="https://www.economist.com/help/about-us"
  >
    About us
    <span
      className={undefined}
    >
      <svg
        version="1.1"
        viewBox="0 0 48 48"
        x="0px"
        xmlSpace="preserve"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <polygon
          points="17.6,14.8 20.4,12 32.4,24 20.4,36 17.6,33.2 26.7,24 	"
        />
      </svg>
    </span>
  </a>
</li>
`;
